File: docs/languages/java/MyApp.java
public class MyApp {
    public static void main(String[] args) throws Exception {
        Lightbulb bulb = new Lightbulb();

        bulb.start();

        System.out.println("Press <enter> to toggle the light switch.");
        System.out.println("Press ^C to quit.");

        while(true) {
            System.in.read();
            bulb.dispatchEvent(Lightbulb.EventId.SWITCH);
        }
        
    }
}

File: docs/languages/csharp/MyApp.cs

public class MyApp
{
    
    public static void Main()
    {
        LightbulbCallback callback = new LightbulbCallback();
        Lightbulb bulb = new Lightbulb(callback);
        bulb.Start();

        Console.WriteLine("Press <enter> to toggle the light switch.");
        Console.WriteLine("Press ^C to quit.");

        while (true) {            
            Console.ReadLine();
            bulb.DispatchEvent(Lightbulb.EventId.SWITCH);
        }
    }
}